app/src/main/java/org/oppia/android/app/topic/TopicFragmentPresenter.kt

```diff
@@ -78,15 +82,24 @@ class TopicFragmentPresenter @Inject constructor(
 
   private fun setUpViewPager(viewPager2: ViewPager2, topicId: String, isConfigChanged: Boolean) {
     val adapter =
-      ViewPagerAdapter(fragment, internalProfileId, topicId, storyId, enablePracticeTab)
+      ViewPagerAdapter(
+        fragment,
+        internalProfileId,
+        topicId,
+        storyId,
+        enablePracticeTab,
+        myDownloadsFeatureFlag.get().value
+      )
     viewPager2.adapter = adapter
-    TabLayoutMediator(tabLayout, viewPager2) { tab, position ->
-      val topicTab = TopicTab.getTabForPosition(position, enablePracticeTab)
-      tab.text = fragment.getString(topicTab.tabLabelResId)
-      tab.icon = ContextCompat.getDrawable(activity, topicTab.tabIconResId)
-    }.attach()
-    if (!isConfigChanged && topicId.isNotEmpty()) {
-      setCurrentTab(if (storyId.isNotEmpty()) TopicTab.LESSONS else TopicTab.INFO)
+    if (!myDownloadsFeatureFlag.get().value) {
```

Comment:
Will this break the post-download flow? The tabs should still work in that case.

Reply:
fixed